### PR TITLE
Add WebController and /web/welcome site.

### LINKED ISSUE #7: Add MVC-controller.

### PR DESCRIPTION
**Pull request by Win-ther**
### Related Issue(s)
Closes #7

### Proposed Changes
- Added WebController, html-file, css-file and js-file for webpage web/welcome.
- Add pagination for public messages.
- Add initialization for database.

### Description
We added a way to see all our public messages and pagination on /welcome.

### CODE REVIEW

File: src/main/java/se/iths/springbootgroupproject/SpringbootGroupProjectApplication.java [hunk 1]
```diff
@@ -11,5 +21,62 @@ public class SpringbootGroupProjectApplication {
 	public static void main(String[] args) {
 		SpringApplication.run(SpringbootGroupProjectApplication.class, args);
 	}
-
+	@Bean
+	@Order(Ordered.HIGHEST_PRECEDENCE)
```
Comment (Frlundsten) [inline src/main/java/se/iths/springbootgroupproject/SpringbootGroupProjectApplication.java]:
```suggestion
	@Order(Ordered.HIGHEST_PRECEDENCE)
```
Is there a reason we use this now? Data loads fine without it and I cannot find another @Order

File: src/main/resources/templates/welcome.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+</head>
+<body>
+<div class="dropdown">
+    <i class="gg-globe-alt" id="globe-icon"></i>
+    <div class="dropdown-content" id="dropdown-content">
+        <a th:href="@{${httpServletRequest.requestURI} + '?page=' + ${currentPage} + '&lang=en'}">English</a>
+        <a th:href="@{${httpServletRequest.requestURI} + '?page=' + ${currentPage} + '&lang=sv'}">Svenska</a>
+    </div>
+</div>
+<script src="/static/script.js"></script>
```
Comment (Frlundsten) [inline src/main/resources/templates/welcome.html]:
```suggestion
<script src="/static/script.js"></script>
```
I think we should place it in header since it is part of the setup and right now it is mixed with the content, makes no difference but I do think it is cleaner to keep the setup from the content.  ```<script src="/static/script.js" defer></script>``` We should use defer keyword to improve performance. [Defer](https://javascript.info/script-async-defer#defer:~:text=The%20defer%20attribute%20tells,but%20with%20defer%3A)

File: src/main/resources/static/script.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// JavaScript för att visa dropdown-rutan när användaren klickar på ikonen
+document.getElementById("globe-icon").addEventListener("click", function() {
+    var dropdownContent = document.getElementById("dropdown-content");
```
Comment (Frlundsten) [inline src/main/resources/static/script.js]:
```suggestion
    let dropdownContent = document.getElementById("dropdown-content");
```
I do not think we need to reach the variable globally. Improves encapsulation with let.

File: src/main/resources/static/styles.css [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    display: block;
+}
+
+/* Main table styles */
+.styled-table {
+    border-collapse: collapse;
+    margin: 5px auto;
+    font-size: 0.9em;
+    font-family: sans-serif;
+    min-width: 800px;
```
Comment (Frlundsten) [inline src/main/resources/static/styles.css]:
```suggestion
    width: 80vw; (or whatever value that looks best )
```
I think we should use vw as width because it is determined by the viewport. Making it look better on different platforms.

File: src/main/resources/templates/welcome.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>Welcome to GilfGang website</title>
+        <link href='https://unpkg.com/css.gg@2.0.0/icons/css/globe-alt.css' rel='stylesheet'>
+        <link href='https://unpkg.com/css.gg@2.0.0/icons/css/arrow-right-r.css' rel='stylesheet'>
+        <link href='https://unpkg.com/css.gg@2.0.0/icons/css/arrow-left-r.css' rel='stylesheet'>
+    <link rel="stylesheet" type="text/css" href="/static/styles.css">
```
Comment (Frlundsten) [inline src/main/resources/templates/welcome.html]:
```suggestion
    <link rel="stylesheet" href="/static/styles.css">
```
Why do you need type="text/css" ?

File: src/main/java/se/iths/springbootgroupproject/controllers/WebController.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * OM TID FINNS-> Alla användare sida: kunna se alla användare och söka på användare
+     */
+    private final MessageService messageService;
+    private final UserService userService;
+
+    public WebController(MessageService messageService, UserService userService) {
+        this.messageService = messageService;
+        this.userService = userService;
+    }
```
Comment (Frlundsten) [inline src/main/java/se/iths/springbootgroupproject/controllers/WebController.java]: UserService is unused, should be removed.

**Review by Frlundsten — CHANGES_REQUESTED** [review]
Well done with the massive work, have a few questions though. 

Why do we not use htmx?

**Review by marcusju23 — DISMISSED** [review]
Great job overall! 🤯 I got one thing I'd like to point out.

The current `databaseInit` method repeats code for message and user creation, hindering maintenance and violating the [DRY](https://docs.getdbt.com/terms/dry) principle. 

I suggest adding methods in `MessageService` and optionally `UserService` for better maintainability and adherence to best practices. This could potentially be addressed as a separate issue or resolved within the current PR.

Check [SonarCloud](https://sonarcloud.io/component_measures?metric=new_duplicated_lines_density&selected=Win-ther_SpringbootGroupProject%3Asrc%2Fmain%2Fjava%2Fse%2Fiths%2Fspringbootgroupproject%2FSpringbootGroupProjectApplication.java&view=list&pullRequest=44&id=Win-ther_SpringbootGroupProject) for better highlightning.


### **TL;DR: Address repeating code directly or create a separate issue for it. Approved regardless! 😄**

**Comment by altmannk** [comment]
> Great job overall! 🤯 I got one thing I'd like to point out.
> 
> The current `databaseInit` method repeats code for message and user creation, hindering maintenance and violating the [DRY](https://docs.getdbt.com/terms/dry) principle.
> 
> I suggest adding methods in `MessageService` and optionally `UserService` for better maintainability and adherence to best practices. This could potentially be addressed as a separate issue or resolved within the current PR.
> 
> Check [SonarCloud](https://sonarcloud.io/component_measures?metric=new_duplicated_lines_density&selected=Win-ther_SpringbootGroupProject%3Asrc%2Fmain%2Fjava%2Fse%2Fiths%2Fspringbootgroupproject%2FSpringbootGroupProjectApplication.java&view=list&pullRequest=44&id=Win-ther_SpringbootGroupProject) for better highlightning.
> 
> ### **TL;DR: Address repeating code directly or create a separate issue for it. Approved regardless! 😄**

We've placed a comment just to clarify 🤓 "//For development purpose we init db with some values"

**Comment by marcusju23** [comment]
> We've placed a comment just to clarify 🤓 "//For development purpose we init db with some values"

Yes, I'm aware. However, I believe this could have been achieved through testing. Alternatively create a test branch or something similar. Pushing code to the main branch that we know will be deleted later feels redundant and a bit wrong, in my opinion.

**Comment by Win-ther** [comment]
> 
> Yes, I'm aware. However, I believe this could have been achieved through testing. Alternatively create a test branch or something similar. Pushing code to the main branch that we know will be deleted later feels redundant and a bit wrong, in my opinion.

To actually use our project, maybe we should have initalizers for our database with a couple of messages? Since the db from start will be empty. This should be done better though as you said and we could put up an Issue for this to do better later since it is not integral for the project right now. How we have it right now is fine for the moment.

**Review by Frlundsten — APPROVED** [review]
Good work guys!

**Review by marcusju23 — APPROVED** [review]
i-mean-its-alright

**Review by altmannk — APPROVED** [review]
WOW THIS IS AMAZING! 😍
